### PR TITLE
Build: Remove dist on commit, add dist on release

### PR DESCRIPTION
**Pull request by omgaz**
- Only lint staged files (like adslot ui)
- Only run dist on release
- On release, pre-dist, make sure all node modules are installed.
- Add check that you're on master for release
- Automatically pull pre-release

### CODE REVIEW

**Comment by vinteo** [comment]
I wonder if we should do the non minified one still on commit?

**Comment by omgaz** [comment]
@vinteo that's only in adslot-ui

File: scripts/git-hooks/pre-commit [hunk 1]
```diff
@@ -8,30 +8,72 @@ then
...
 then
-  echo "Failed, attempting to commit to master. Please create a branch for your work." && exit 1
+  echo "💔  Cannot commit to master, please create a branch." && exit 1
 fi
 
 
 # Run linting and commit check.
-echo "  - linting"
-npm run posttest
+STAGED_SRC_FILES=`git diff --staged --name-only HEAD ./src | grep .*\\.js | grep -v json`; # JS & JSX Files included
```
Comment (jharrington22) [inline scripts/git-hooks/pre-commit]: Semicolon's are redundant and should be removed

File: scripts/git-hooks/pre-commit [hunk 2]
```diff
@@ -8,30 +8,72 @@ then
...
+
+if [ $SRC_EXIT_CODE -eq 1 ] || [ $TEST_EXIT_CODE -eq 1 ] || [ $SASS_EXIT_CODE -eq 1 ] || [ $JSCS_EXIT_CODE -eq 1 ]; then
+  echo "💔  Linting failed.";
+  exit 1;
+fi
 
-echo "…done."
+echo "🔍  Checking commit message.";
+node scripts/commit-msg && COMMIT_MSG_EXIT_CODE=0;
+if [ $COMMIT_MSG_EXIT_CODE -eq 1 ]; then exit 1; fi
```
Comment (jharrington22) [inline scripts/git-hooks/pre-commit]: Should be multiple lines to match the rest. You could shorten to `[ $COMMIT_MSG_EXIT_CODE ] && exit 1` if you wanted to.

File: scripts/git-hooks/pre-commit [hunk 3]
```diff
@@ -8,30 +8,72 @@ then
...
+SRC_EXIT_CODE=1;
+TEST_EXIT_CODE=1;
+SASS_EXIT_CODE=1;
+JSCS_EXIT_CODE=1;
+COMMIT_MSG_EXIT_CODE=1;
 
+echo "🔍  Running linting on staged files…";
+echo "    - ESLint Source";
+if [ "$STAGED_SRC_FILES" ]; then
+  eslint --rulesdir ./src $STAGED_SRC_FILES && SRC_EXIT_CODE=0;
```
Comment (jharrington22) [inline scripts/git-hooks/pre-commit]: Since `$STAGED_SRC_FILES` is an array you should expand it with `${STAGED_SRC_FILES[@]}`

Comment (jharrington22) [inline scripts/git-hooks/pre-commit]: Same for the others `STAGED_TEST_FILES` and `STAGED_SASS_FILES`

**Comment by vinteo** [comment]
Merging...

**Pull request merged**